src/openrct2/peep/peep.c

```diff
@@ -8321,36 +8341,56 @@ static sint32 peep_interact_with_path(rct_peep* peep, sint16 x, sint16 y, rct_ma
 
 		uint8 rideIndex = map_element->properties.path.ride_index;
 
+		// Check if this queue path tile is connected to a ride.
 		if (rideIndex == 0xFF){
+			// Queue is not connected to a ride.
 			peep->interactionRideIndex = 0xFF;
 			return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 		}
 
+		// Queue is connected to a ride.
 		if (peep->state == PEEP_STATE_QUEUING){
+			// Check if this queue is connected to the ride the
+			// peep is queuing for, i.e. the player hasn't edited
+			// the queue.
 			if (peep->current_ride == rideIndex){
 				return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 			}
+			// Queue got disconnected from the ride.
 			remove_peep_from_queue(peep);
 			peep_decrement_num_riders(peep);
 			peep->state = PEEP_STATE_1;
 			peep_window_state_update(peep);
 			return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 		}
 
+		// One of two cases applies here:
+		// 1. Peep is in the ride queue but no longer queuing.
+		//    This usually means the ride was closed.
+		// 2. Peep is walking up to the queue entrance from
+		//    a normal path tile.
+
 		if (peep->interactionRideIndex == rideIndex){
+			// Case 1 above applies, so walk the queue.
 			return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 		}
 
+		// Case 2 above applies - decide whether to go on the ride.
 		peep->var_F4 = 0;
 		uint8 stationNum = (map_element->properties.path.additions & 0x70) >> 4;
 		if (!peep_should_go_on_ride(peep, rideIndex, stationNum, PEEP_RIDE_DECISION_AT_QUEUE)){
-			peep->interactionRideIndex = rideIndex;
+			// Peep has decided not to go on the ride.
```

Comment:
It's inconsistent (in my opinion).
In some places it is used like this ... although it's effect is limited to within a single path tile. As soon as a peep walks into another tile it's reset. Problem is it's used inconsistently for this purpose.  E.g. Interacting with a shop doesn't reset it, so bouncing off a shop back onto the ride queue results in the peep walking up the queue.
 
The main (and more important) use of interactionRideIndex is to let the peep walk up the queue when actually queuing and also and walk out of the queue when not queuing.

This particular line (8347 in the original) is critical for letting the peep walk back out of the queue when they aren't queuing. Without it they get stuck in the queue. Removing this fixed peeps walking into queues when they shouldn't, but trapped peeps trying to get back out of a queue - quite a regression.
A much improved fix is in #5442 commit ef1bdba - it probably shouldn't been part of that PR, but that's where the above regression was identified.